File: hellohilt/src/main/java/com/airbnb/mvrx/hellohilt/di/AppModule.java
package com.airbnb.mvrx.hellohilt.di;

import com.airbnb.mvrx.hellohilt.HelloViewModel;
import com.airbnb.mvrx.hellohilt.HelloViewModel_AssistedFactory;

import dagger.Binds;
import dagger.Module;
import dagger.hilt.InstallIn;
import dagger.hilt.android.components.ApplicationComponent;
import dagger.multibindings.IntoMap;

@Module
@InstallIn(ApplicationComponent.class)
public abstract class AppModule {

    @Binds
    @IntoMap
    @ViewModelKey(HelloViewModel.class)
    abstract AssistedViewModelFactory<?, ?> helloViewModelFactory(HelloViewModel_AssistedFactory factory);
}

File: hellohilt/src/main/java/com/airbnb/mvrx/hellohilt/di/AppModule.kt
package com.airbnb.mvrx.hellohilt.di

import com.airbnb.mvrx.hellohilt.HelloViewModel
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import dagger.multibindings.IntoMap

/**
 * The InstallIn SingletonComponent scope must match the context scope used in [HiltMavericksViewModelFactory]
 *
 * If you want an Activity or Fragment scope
 */
@Module
@InstallIn(SingletonComponent::class)
interface AppModule {

    @[Binds IntoMap ViewModelKey(HelloViewModel::class)]
    fun helloViewModelFactory(factory: HelloViewModel.Factory): AssistedViewModelFactory<*, *>
}
